src/chocolatey/infrastructure/guards/SingleGlobalMutex.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright © 2017 Chocolatey Software, Inc
+// Copyright © 2011 - 2017 RealDimensions Software, LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 
+// You may obtain a copy of the License at
+// 
+// 	http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace chocolatey.infrastructure.guards
```

Comment:
This feels more like the synchronization namespace, next to mutexes and semaphores. I just looked and we don't have those in the Chocolatey source code...

Here is what they look like: https://github.com/chocolatey/package-validator/tree/master/src/chocolatey.package.validator/infrastructure/synchronization